### PR TITLE
ESPN

### PR DESCRIPTION
**Pull request by EatDatPussy069**

### CODE REVIEW

**Comment by EatDatPussy069** [comment]
Screenshot's of the ESPN Presence Working,

Link's  -
 
https://gyazo.com/f18ca7f59e24bb490f781397db2cc685
image

https://gyazo.com/988db9465065b9f80ed3ffaa8f653d3a
image

https://i.gyazo.com/06c5df650a84dc63cfd949208f59afd3.png
image

image

File: programs/E/ESPN/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "author": {
+    "name": "𝐃 𝐘 𝐋 𝐀 𝐍",
+    "id": "315487546362560513"
+  },
+  "contributors": [{
+    "name": "𝐃 𝐘 𝐋 𝐀 𝐍",
+    "id": "315487546362560513"
+  }],
```
Comment (Bas950) [inline programs/E/ESPN/dist/metadata.json:9]: why did you add yourself as contributor xD

File: programs/E/ESPN/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "service": "ESPN",
+  "description": {
+    "en": "Worldwide leader in sports forever"
+  },
+  "url": "www.espn.com",
+  "version": "1.0",
```
Comment (Bas950) [inline programs/E/ESPN/dist/metadata.json:15]: version 1.0.0

File: programs/E/ESPN/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "logo": "https://i.imgur.com/k7urXfW.png",
+  "thumbnail": "https://i.imgur.com/TgsuZOb.jpg",
+  "color": "#D61E26",
+  "tags": ["SPORTS", "ESPN"],
+  "category": "Other"
```
Comment (Bas950) [inline programs/E/ESPN/dist/metadata.json:20]: tags and category in lowercase

File: programs/E/ESPN/presence.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    // element grabs //
+    // api calls //
+    // variable sets //
+}
+
+setInterval(10000, myOutsideHeavyLiftingFunction);
+//Run the function seperate from the UpdateData event every 10 seconds to get and set the variables which UpdateData picks up
+
+*/
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:18]: remove comments

File: programs/E/ESPN/presence.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+var browsingStamp = Math.floor(Date.now() / 1000);
+var teamname;
+presence.on("UpdateData", async () => {
+    /*UpdateData is always firing, and therefore should be used as your refresh cycle, or `tick`. This is called several times a second where possible.
+
+    It is recommended to set up another function outside of this event function which will change variable values and do the heavy lifting if you call data from an API.*/
+
+    var presenceData: presenceData = {
+        largeImageKey: "espnapp_logo" /*The key (file name) of the Large Image on the presence. These are uploaded and named in the Rich Presence section of your application, called Art Assets*/
+    }; /*Optionally you can set a largeImageKey here and change the rest as variable subproperties, for example presenceSata.type = "blahblah"; type examples: details, state, etc.*/
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:36]: remove comments

File: programs/E/ESPN/presence.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    } 
+    if (presenceData.details == null) {
+        //This will fire if you do not set presence details
+        presence.setTrayTitle(); //Clears the tray title for mac users
+        presence.setActivity(); /*Update the presence with no data, therefore clearing it and making the large image the Discord Application icon, and the text the Discord Application name*/
+    } else {
+        //This will fire if you set presence details
+        presence.setActivity(presenceData); //Update the presence with all the values from the presenceData object
+    }
+});
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:126]: remove comments

File: programs/E/ESPN/presence.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+    else if (document.location.pathname.includes("/search/")){
+        var search = (document.querySelector("#fittPageContainer > div.page-container.cf > div > div > div.SearchBar.ml4.mr4 > form > input")as HTMLInputElement).value;
+        presenceData.details = "searching";
+        presenceData.state = search;
+        presenceData.smallImageKey = "search";
+        presenceData.smallImageText = "searching";
+        console.log (search);
+    }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:117]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                presenceData.smallImageText = "playing";
+            }
+            else {
+                delete presenceData.startTimestamp;
+                delete presenceData.endTimestamp;
+                presenceData.smallImageKey = "pause";
+                presenceData.smallImageText = "paused";
+            }
+            console.log(timestamps);
+        }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:103]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+            presenceData.state = "Stats " + stats;
+            console.log("Stats Page");
+        }
+        else {
+            teamname = document.querySelector("#clubhouse-header > div > div > div > h1 > div").textContent;
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = teamname;
+            presenceData.state = "Home Page";
+            console.log("teampage"); 
+        }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:75]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    else if(document.location.pathname.includes("/nfl/team/")){
+        if(document.location.pathname.includes("/nfl/team/stats/")){
+            teamname = document.querySelector("#fittPageContainer > div.StickyContainer > div:nth-child(1) > div > div > div > div > h1 > div").textContent;
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = teamname;
+            stats = document.querySelector("#fittPageContainer > div.StickyContainer > div.page-container.cf > div.layout.is-9-3 > div > section > div > div.flex.justify-between.mt3.mb3.items-center > h1").textContent;
+            stats = stats.replace(/[^\d-]/g,``);
+            presenceData.state = "Stats " + stats;
+            console.log("Stats Page");
+        }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:68]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+        else {
+            teamname = document.querySelector("#clubhouse-header > div > div > div > h1 > div").textContent;
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = teamname;
+            presenceData.state = "Home Page";
+            console.log("teampage"); 
+        }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:57]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    else if(document.location.pathname.includes("/nba/team/")){
+        if(document.location.pathname.includes("/nba/team/stats/")){
+            teamname = document.querySelector("#fittPageContainer > div.StickyContainer > div:nth-child(1) > div > div > div > div > h1 > div").textContent;
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = teamname;
+            stats = document.querySelector("#fittPageContainer > div.StickyContainer > div.page-container.cf > div.layout.is-9-3 > div > section > div > div.flex.justify-between.mt3.mb3.items-center > h1").textContent;
+            stats = stats.replace(/[^\d-]/g,``);
+            presenceData.state = "Stats " + stats;
+            console.log("Stats Page");
+        }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:50]: remove console.log

File: programs/E/ESPN/presence.ts [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+        presence.setActivity(); /*Update the presence with no data, therefore clearing it and making the large image the Discord Application icon, and the text the Discord Application name*/
+    } else {
+        //This will fire if you set presence details
+        presence.setActivity(presenceData); //Update the presence with all the values from the presenceData object
+    }
+});
+
+//document.querySelector("#fittPageContainer > div.WatchListingsVideo.WatchListingsVideo--VideoContainer > div.WatchVideoPlayer__Metadata > div.WatchVideoPlayer__Metadata--title").textContent
+
+//document.querySelector("#vjs_video_3")
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:130]: why these comments?

File: programs/E/ESPN/dist/metadata.json [hunk 14]
```diff
@@ -0,0 +1,21 @@
+{
+  "author": {
+    "name": "𝐃 𝐘 𝐋 𝐀 𝐍",
+    "id": "315487546362560513"
+  },
+  "contributors": [{
+    "name": "",
+    "id": ""
+  }],
```
Comment (Bas950) [inline programs/E/ESPN/dist/metadata.json:9]: just remove it, its not required

File: programs/E/ESPN/presence.ts [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+    return [Math.floor(startTime / 1000),Math.floor(endTime)];
+  }
+
+
+  
+var browsingStamp = Math.floor(Date.now() / 1000);
+var teamname;
+presence.on("UpdateData", async () => {
+
+    var presenceData: presenceData = {
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:18]: please use let and const instead of var

File: programs/E/ESPN/presence.ts [hunk 16]
```diff
@@ -0,0 +1,106 @@
+var presence = new Presence({
+    clientId: "707379503881650258"
+});
+var stats;
+
+function getTimestamps(videoTime: number, videoDuration: number): any {
+    var startTime = Date.now();
+    var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+    return [Math.floor(startTime / 1000),Math.floor(endTime)];
+  }
```
Comment (Bas950) [inline programs/E/ESPN/presence.ts:10]: Dont return any
use: `Array<number>`

**Comment by Bas950** [comment]
You also didnt remove the console.logs, please remove them.

File: programs/E/ESPN/presence.ts (hunk 16 above)
Comment (EatDatPussy069) [reply programs/E/ESPN/presence.ts]: Fixed New Commit

File: programs/E/ESPN/dist/metadata.json (hunk 14 above)
Comment (EatDatPussy069) [reply programs/E/ESPN/dist/metadata.json]: Fixed Commit

**Comment by Bas950** [comment]
Will close in 3 days if no changes were made
